libkbfs/keybase_daemon_rpc.go

```diff
@@ -305,3 +306,46 @@ func (k *KeybaseDaemonRPC) Shutdown() {
 		k.shutdownFn()
 	}
 }
+
+// GetTLFCryptKeys implements the TlfKeysInterface interface for
+// KeybaseDaemonRPC.
+func (k *KeybaseDaemonRPC) GetTLFCryptKeys(
+	ctx context.Context, tlfName string) (res keybase1.TLFCryptKeys, err error) {
+	var tlfHandle *TlfHandle
+
+getHandle:
+	for {
+		tlfHandle, err = ParseTlfHandle(ctx, k.config.KBPKI(), tlfName, false)
+		switch e := err.(type) {
+		case TlfNameNotCanonical:
+			tlfName = e.NameToTry
+		case nil:
+			break getHandle
+		default:
+			return res, err
+		}
+	}
+
+	res.CanonicalName = keybase1.CanonicalTlfName(tlfHandle.GetCanonicalName())
+
+	var rmd ImmutableRootMetadata
+	var id TlfID
+	_, rmd, id, err = k.config.KBFSOps().GetOrInitializeNewMDMaster(
+		ctx, k.config.MDOps(), tlfHandle)
+	if err != nil {
+		return res, err
+	}
+	res.TlfID = keybase1.TLFID(id.String())
```

Comment:
Why do we need to return the TLF ID?  (Sorry, I'm missing some context.)  If the caller can get this some other way (say from the API call that returns the favorites), then we could take that as an input parameter instead of a name, and avoid the TLF handle all together.